### PR TITLE
Pytorch unet tariq

### PR DESCRIPTION
**Pull request by tkalim**
- Saves checkpoints every 5 epochs
- Can run on gpu if cuda is available
- Can be run alone with `python train.py`
- Warning: not tested, because I don't know the right params for the dataset, but should be ok

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -68,3 +75,17 @@ def train(
                 model=model, epoch=epoch, loss=loss.item(), save_dir=checkpoints_dir
             )
             print(f"model saved to {str(checkpoints_dir)}")
+
+
+if __name__ == "__main__":
+    model = UNet()
+    # TODO: Put right init params in dataset
```
Comment (Kaabachi) [inline train.py]: params we are currently using 

PATCH_SIZE = 16
LARGE_PATCH_SIZE = 96

TRAIN_IMAGE_INITIAL_SIZE = 400

File: predict.py [hunk 2]
```diff
@@ -1,58 +1,72 @@
...
+        if cuda:
+            batch_images = batch_images.to(device="cuda")
+        with torch.no_grad():
+            output = model(batch_images)
+
+        final = output[0][0].clone().detach().cpu()
+        final[final > 0.5] = 1
+        final[final <= 0.5] = 0
 
-        final = output[0][0].detach()
```
Comment (Kaabachi) [inline predict.py:32]: Personellement j'avais fait final = output[0][0].cpu().detach(), on peut detach avant de le mettre en cpu ? ca ne renvoie pas d'erreur ?

File: predict.py [hunk 3]
```diff
@@ -1,58 +1,72 @@
...
-    
+    prediction_data_dir = "./Datasets/predictions/"
+
     mask = image.clone().detach()
-    
+
     img = transforms.ToPILImage()(mask)
-    img.save(prediction_data_dir+"img" + str(i) + ".png","PNG")
+    img.save(prediction_data_dir + "img" + str(i) + ".png", "PNG")
+
```
Comment (Kaabachi) [inline predict.py:21]: devrait être str(i+1) je changerai après
